tasks/contract.ts

```diff
@@ -95,3 +97,46 @@ async function generate({ x, y }: { x: number; y: number }, hre: HardhatRuntimeE
 
   return ret;
 }
+
+task("request:coords", "request a reveal")
+  .addParam("x", "x value", undefined, types.int)
+  .addParam("y", "y value", undefined, types.int)
+  .setAction(requestCoords);
+
+async function requestCoords({ x, y }: { x: number; y: number }, hre: HardhatRuntimeEnvironment) {
+  const BroadcastMarketFactory = await hre.ethers.getContractFactory("BroadcastMarket");
+  const broadcastMarket = BroadcastMarketFactory.attach(BROADCAST_MARKET_ADDRESS);
+
+  const overrides = {
+    value: hre.REQUEST_MINIMUM,
+  };
+
+  const proof: RevealSnarkContractCallArgs = await hre.run("generate", { x, y });
+
+  const revealRequestTx = await broadcastMarket.requestReveal(...proof, overrides);
+  await revealRequestTx.wait();
+
+  console.log("requested"); // todo add more logging
+  return;
+}
+
+task("request:locationId", "request a reveal")
+  .addPositionalParam("locationId", "0 padded hex value of locationDec, no 0x prefix", undefined, types.string)
+  .setAction(requestLocationId);
+
+async function requestLocationId({ locationId }: { locationId: string }, hre: HardhatRuntimeEnvironment) {
+  const BroadcastMarketFactory = await hre.ethers.getContractFactory("BroadcastMarket");
+  const broadcastMarket = BroadcastMarketFactory.attach(BROADCAST_MARKET_ADDRESS);
+
+  const overrides = {
+    value: hre.REQUEST_MINIMUM,
+  };
+
+  const locationDecString = locationIdToDecStr(locationId as LocationId);
+
+  const revealRequestTx = await broadcastMarket.requestRevealPlanetId(locationDecString, overrides);
+  await revealRequestTx.wait();
+
+  console.log("requested"); // todo add more logging
```

Comment:
I basically just want to see the request list when this finishes.

Reply:
I can hre.run("list") but that could be pretty big with no filtering for active unpaid requests or something

Reply:
Hm, yeah, idk. But while I was debugging, I needed to know what was there. I'm fine with whatever